salto-io/salto
# SALTO-3609 and SALTO-4032 Added fetch and deploy for appUserSchema and added appUserSchema and userSchema to validtor

1. Added the capability to fetch appUserSchema. 
2. Added the capability to deploy appUserSchema. 
3. Added appUserSchema and userSchema to `schemaFieldRemovalValidator"
4. Added tests to check the validator. 
5. Added appUserSchema to deleteRecurseIntoFilter.

---

_Additional context for reviewer:_
To test the code manually do as follows:
1. add custom property to a user schema. 
2. fetch it to your env.
3. delete the custom property from the nacl. 
4. check that the attribute was deleted in the schema. 
Do the same for appUserSchema.

---
_Release Notes_: 
None

---
_User Notifications_: 
None
